frontend/src/lib/types.ts

```diff
@@ -24,11 +24,7 @@ export const signUpSchema = z
       ),
     confirmPassword: z
       .string()
-      .min(8, 'Password must be at least 8 character')
-      .regex(
-        passwordRegex,
-        'Password must be contains at least one uppercase and one lowercase and one digit and one special character'
-      ),
+      .min(1, { message: 'Confirm Password is required'}),
   })
```

Comment:
Why is 1? and why are we completely changing this? passwordRegex is totally removed?

Reply:
> Why is 1? and why are we completely changing this? passwordRegex is totally removed?

Given 1 so that if it is empty then we can show error message "Confirm Password is required".
And if it has at least 1 charactor in confirmPassword field it should check with password field string that confirmPassword and password field string matches or not so in this case it will show error message "Confirm Password does not match".

And if confirmPassword string matches with password string then it won't show error message.

Password regex is removed because it does not require here. We are already added regex to password field so it is obivous that user will have to fill same password string in confirmPassword. We don't need to tell explicity for this field.

Reply:
Good attention to details!